packages/devtools_app_shared/lib/src/service/service_manager.dart

```diff
@@ -433,7 +433,12 @@ class ServiceManager<T extends VmService> {
 
   /// This can throw an [RPCError].
   Future<void> performHotRestart() async {
-    await callServiceOnMainIsolate(hotRestartServiceName);
+    isolateManager.hotRestartInProgress = true;
+    try {
+      await callServiceOnMainIsolate(hotRestartServiceName);
+    } catch (_) {
+      isolateManager.hotRestartInProgress = false;
+    }
```

Comment:
`isolateManager.hotRestartInProgress` will only be accurate when a hot restart is triggered through DevTools. If a user triggers hot restart through a connected IDE or through the command line, this code will not get hit.

Reply:
This is another example where this would be useful to have: https://github.com/flutter/flutter/issues/134470

Reply:
Added a TODO and a comment